contributor-docs/adrs/adr-XXX-file-structure.md

```diff
@@ -0,0 +1,42 @@
+# ADR XXX: File structure
+
+## Status
+
+Proposed
+
+## Context
+
+Components might be more grokable if they were structured consistently. This ADR proposes conventions
+
+## Decision
+
+TL;DR:
+
+```
+primer-react/
+├─ src/
+│  ├─ BreadCrumbs/
+│  │  ├─ index.ts                    // Just re-exporting?
+│  │  ├─ BreadCrumbs.tsx             // Primary component
+│  │  ├─ BreadCrumbsItem.tsx         // Subcomponent (include item name for findability in most IDEs)
+│  │  ├─ BreadCrumbs.mdx             // Documentation. Always .mdx, not .md
+│  │  ├─ BreadCrumbs.stories.tsx
+│  │  ├─ BreadCrumbs.test.tsx        // Unit tests
+│  │  ├─ BreadCrumbs.types.test.tsx  // Type tests
+│  │  ├─ BreadCrumbs.yml             // Component metadata (Possible future)
+│  │  └─ __snapshots__/
+┆  ┆
+```
+
+### Rules
+
+- Every component should have its own PascalCased directory directly under `src/`
+- Subcomponents meant to be used as children of their parent component should be properties of the exported component (e.g., `BreadCrumbs.Item`)
```

Comment:
Does this mean that the `Object.assign` happens in `Breadcrumbs.tsx` or `index.tsx`?

Reply:
I didn't mean to specify — but if you think this should I'm fine with that!

Reply:
We can add more rules to this doc as we form opinions. So I guess it's fine to be vague now.

IMO, the `Object.assign` should happen in `Breadcrumbs.tsx`. `index.tsx` seems mostly like an implementation detail to make imports nicer

Reply:
Should index.tsx export default or named?

Reply:
Hmm. I lean towards named export. In general, I'm not a big fan of default exports (unless they're necessary for some reason)